smoke-tests/java9-modules/java9-modules.gradle

```diff
@@ -12,60 +12,22 @@ jar {
   }
 }
 
-// If the current JDK version (the one running gradle) is < 9, we need to find a version >= 9
-// to compile this project.  java.gradle creates a map of java executables 
-// called "javaExecutableVersionCache" pulled from the environment.
-// This loops over the cache to find a usable jdk.
-// Since this project is the only one that requires a version above Java 8
-// it's special cased here instead of putting a generic version matcher in java.gradle
-if (JavaVersion.VERSION_1_9.compareTo(JavaVersion.current()) > 0) {
-  def targetJavaHome
+compileMain_java9Java {
+  options.fork = true
+  options.forkOptions.javaHome = file(findJavaHome(JavaVersion.VERSION_1_9))
```

Comment:
@iNikem I can't build with Java 8 anymore, can we make building this module optional? Or should we update CONTRIBUTING doc to require Java 11?

```
* Where:
Build file 'C:\git\opentelemetry-auto-instr-java\smoke-tests\java9-modules\java9-modules.gradle' line: 17

* What went wrong:
A problem occurred evaluating project ':smoke-tests:java9-modules'.
> path may not be null or empty string. path='null'
```

Reply:
This fails because you don't have `JAVA_9_HOME` env variable pointing to java 9 installation. Can you try as a workaround define it in your env with any value? Don't need to have actual java9 installed.

Reply:
This is what I get:

```
$ export JAVA_9_HOME=xyz

$ ./gradlew :smoke-tests:java9-modules:build

FAILURE: Build failed with an exception.

* What went wrong:
A problem occurred evaluating project ':smoke-tests:java9-modules'.
> A problem occurred starting process 'command 'xyz/bin/java''
```

```
$ export JAVA_9_HOME=/c/dev/java/zulu-11

$ ./gradlew :smoke-tests:java9-modules:build

FAILURE: Build failed with an exception.

* Where:
Build file 'C:\git\opentelemetry-auto-instr-java\smoke-tests\java9-modules\java9-modules.gradle' line: 17

* What went wrong:
A problem occurred evaluating project ':smoke-tests:java9-modules'.
> path may not be null or empty string. path='null'
```

```
$ export JAVA_9_HOME=/c/dev/java/zulu-9

$ ./gradlew :smoke-tests:java9-modules:build

Deprecated Gradle features were used in this build, making it incompatible with Gradle 7.0.
Use '--warning-mode all' to show the individual deprecation warnings.
See https://docs.gradle.org/6.3/userguide/command_line_interface.html#sec:command_line_warnings

BUILD SUCCESSFUL in 8s
241 actionable tasks: 241 up-to-date
```

Reply:
Correct, this is indeed as it was "intended" by my change. Not ideal, but works for now, I think.

Reply:
Ok, can you update CONTRIBUTING.md?

Reply:
Yes, will do